torch/utils/tensorboard/pytorch_graph.py

```diff
@@ -0,0 +1,274 @@
+import numpy as np
+import time
+import torch
+
+from collections import OrderedDict
+from distutils.version import LooseVersion
+
+from tensorboard.compat.proto.config_pb2 import RunMetadata
+from tensorboard.compat.proto.graph_pb2 import GraphDef
+from tensorboard.compat.proto.step_stats_pb2 import StepStats, DeviceStepStats, NodeExecStats, AllocatorMemoryUsed
+from tensorboard.compat.proto.versions_pb2 import VersionDef
+
+from .proto_graph import Node_proto
+
+
+methods_OP = ['attributeNames', 'hasMultipleOutputs', 'hasUses', 'inputs',
+              'kind', 'outputs', 'outputsSize', 'scopeName']
+methods_IO = ['node', 'offset', 'uniqueName']  # 'unique' <int> , 'type' <Tensor<class 'torch._C.Type'>>
+
+
+class Node_base(object):
+    def __init__(self, uniqueName=None, inputs=None, scope=None, tensorSize=None, op_type='UnSpecified', attributes=''):
+        self.uniqueName = uniqueName
+        self.inputs = inputs
+        self.tensorSize = tensorSize
+        self.kind = op_type
+        self.attributes = attributes
+        if scope is not None:
+            self.scope = scope
+
+    def __repr__(self):
+        repr = []
+        repr.append(str(type(self)))
+        for m in dir(self):
+            if '__' not in m:
+                repr.append(m + ': ' + str(getattr(self, m)) + str(type(getattr(self, m))))
+        return '\n'.join(repr) + '\n\n'
+
+
+class Node_py(Node_base):
+    def __init__(self, Node_cpp, valid_mothods):
+        super(Node_py, self).__init__(Node_cpp)
+        self.valid_mothods = valid_mothods[:]
+        self.inputs = []
+
+        for m in self.valid_mothods:
+            if m == 'inputs' or m == 'outputs':
+                list_of_node = list(getattr(Node_cpp, m)())
+                io_uniqueName_list = []
+                io_tensorSize_list = []
+                for n in list_of_node:
+                    io_uniqueName_list.append(n.uniqueName())
+                    if n.type().kind() == 'CompleteTensorType':
+                        io_tensorSize_list.append(n.type().sizes())
+                    else:
+                        io_tensorSize_list.append(None)
+
+                setattr(self, m, io_uniqueName_list)
+                setattr(self, m + 'TensorSize', io_tensorSize_list)
+
+            else:
+                setattr(self, m, getattr(Node_cpp, m)())
+
+
+class Node_py_IO(Node_py):
+    def __init__(self, Node_cpp, input_or_output=None):
+        super(Node_py_IO, self).__init__(Node_cpp, methods_IO)
+        try:
+            tensorsize = Node_cpp.type().sizes()
+        except RuntimeError:
+            tensorsize = [1, ]  # fail when constant model is used.
```

Comment:
What's going on here? In general I don't like silently swallowing errors unless there is good justification, and I don't see the justification here.